docs/sovereign/from-sovereign.md

```diff
@@ -0,0 +1,111 @@
+    pub token_identifier: TokenIdentifier<M>,
+    pub token_nonce: u64,
+    pub token_data: EsdtTokenData<M>,
+}
+```
+
+This struct describes a single token transfer action within an Operation. Each Operation can have one or more of such payments, with that enabling the transfer of a variety of tokens during a cross-chain transaction.
+
+- `token_identifier`: used for the identification of the token
+- `token_nonce`: if the token is Non-Fungible or Semi-Fungible, it will have a custom nonce, if not the value will be 0
+- `token_data`: a structure holding metadata and additional token-related details used for minting, token creation (for more details about this structure, please visit the official ESDT Documentation)
+
+```rust
+pub struct OperationData<M: ManagedTypeApi> {
+    pub op_nonce: TxId,
+    pub op_sender: ManagedAddress<M>,
+    pub opt_transfer_data: Option<TransferData<M>>,
+}
+```
+
+`OperationData` encapsulates the needed information for the *Operation* that needs to be executed. This isn’t just another data definition, we’ve already seen data-related fields elsewhere. Instead, it centralizes the contextual information that *Operation* needs before, during, and after execution.
+
+- `op_nonce`: is used for the identification of each *Operation*
+- `op_sender`: represents the original sender of the *Operation*
+- `opt_transfer_data`: an optional `TransferData` field, when present, contains details about the cross-chain execution of another Smart Contract
+
+```rust
+pub struct TransferData<M: ManagedTypeApi> {
+    pub gas_limit: GasLimit,
+    pub function: ManagedBuffer<M>,
+    pub args: ManagedVec<M, ManagedBuffer<M>>,
+}
+```
+
+`TransferData` represents the description of the remove execution of another Smart Contract, one standout feature of the `executeBridgeOps` endpoint is the capability of executing multiple endpoint calls in one transaction.
+
+- `gas_limit`: specifies the needed gas for the execution of all other endpoints.
+- `function`: the name of the endpoint that will be executed.
+- `args`: the arguments for the calls.
+
+
+:::note
+The source code for the endpoint can be found [here](https://github.com/multiversx/mx-sovereign-sc/blob/main/esdt-safe/src/from_sovereign/transfer_tokens.rs)
+:::
+### Executing an *Operation*
+
+This endpoint serves as the key component in orchestrating cross-chain operations within the bridging mechanism.
+```rust
+#[endpoint(executeBridgeOps)]
+ fn execute_operations(
+    &self, 
+    hash_of_hashes: ManagedBuffer, 
+    operation: Operation<Self::Api>
+)
+```
+- `hash_of_hashes`: the key for the hashes map used to track executed and non executed Operations
+- `operation`: the data and the behavior for the cross-chain execution that must happen
+
+1. If the current chain isn’t a Sovereign Chain, execution must happen inside the Main Chain.
+2. Verify that the given Operation’s hash is registered by the Header-Verifier Smart Contract.
```

Comment:
Just explain what happens in the `execute_operations` endpoint
- calculate operation hash
- verify if registered and lock operation hash
- mint tokens or get from account
- distribute tokens
- emit confirmation event and fail event if needed